cmd/serverless/main.go

```diff
@@ -180,13 +171,17 @@ func runAgent(stopCh chan struct{}) (daemon *serverless.Daemon, err error) {
 	// api key reading
 	// ---------------
 
+	// API key reading priority:
+	// KSM > Secrets Manager > Plaintext API key
+	// If one is set but failing, the next will be tried
+
 	// some useful warnings first
 
 	var apikeySetIn = []string{}
 	if os.Getenv(kmsAPIKeyEnvVar) != "" {
```

Comment:
we read twice this env variable (once in api_key.go + once here) not sure about how performance is affected but we may want to read it only once if possible ? What do you think ?

Reply:
Personally I'm not too worried about it -- this only happens once (on cold start) and I doubt the penalty for reading an environment variable would be noticeable/worth optimizing at this point.